### PR TITLE
hgroup element mapping update

### LINKED ISSUE #123: Change mapping of h1-h6 and hgroup
Provided implementers are on board with https://github.com/whatwg/html/pull/3499 as a way of fixing https://github.com/whatwg/html/issues/83#issuecomment-359871505, the mappings for these elements will need to change as well. This should be rather easy to do for these elements on top of the "heading level" concept they will get once that PR lands.

I figured I'd raise this somewhat early so a) we don't forget about it and b) I might be able to get more feedback this way.

### PR DESCRIPTION
**Pull request by scottaohara**
Related to https://github.com/whatwg/html/pull/7829.

`hgroup` was revised to no longer have any expectation of multiple headings as nested children.  This PR simply revises the mapping of `hgroup` from a generic to a `role=group`. This mapping change is done to indicate this element represents a group of related content, and to allow authors to name this group without the need of a specific ARIA attribute.

Previous discussion about revising the roles of errant headings, beyond the first valid instance, have been put on hold for now, and are not part of this PR.

this resolves #123
this may well help resolving https://github.com/w3c/aria-practices/issues/1157

Implementation bugs filed:
 * [x] WebKit (https://bugs.webkit.org/show_bug.cgi?id=253595)
 * [x] Chromium (https://bugs.chromium.org/p/chromium/issues/detail?id=1422741)
 * [x] Gecko (https://bugzilla.mozilla.org/show_bug.cgi?id=1821156)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/pull/398.html" title="Last updated on Mar 8, 2023, 9:09 PM UTC (d63b6f3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/398/256d57e...d63b6f3.html" title="Last updated on Mar 8, 2023, 9:09 PM UTC (d63b6f3)">Diff</a>

### CODE REVIEW

**Comment by scottaohara** [comment]
marked as not ready for merge until discussion topics are covered and continued work on the associated html issue

**Comment by scottaohara** [comment]
Resolution from today's call - don't do anything regarding `hgroup` role  or `p` roles until HTML spec update lands and there is an explicit desire/need for these additional changes.

**Comment by scottaohara** [comment]
@jnurthen shall we talk about this again now that the updated heading outline has landed in HTML?

**Review by stevefaulkner — APPROVED** [review]

**Comment by scottaohara** [comment]
@jnurthen, spoke to @stevefaulkner this morning.  

agreed that, especially in lieu of updating the mapping for `address`, it makes sense to map `hgroup` to `group`.
Otherwise, for now at least - leave the idea of a subheading/subtitle role alone.  We can always re-assess in the future if people actually "want" a role like this announced/exposed. 

Extra (now) invalid headings within the hgroup just need to be decided upon.  If they remain headings, or if they get remapped.

File: index.html [hunk 1]
```diff
@@ -1197,12 +1197,16 @@ <h3>HTML Element Role Mappings</h3>
...
+              <td class="aria"><a class="core-mapping" href="#role-map-group">`group`</a> role </td>
               <td class="ia2"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="uia"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="atk"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="ax"><div class="general">Use WAI-ARIA mapping</div></td>
-              <td class="comments"></td>
+              <td class="comments">
+                If an `hgroup` contains multiple heading elements, then the heading element with the highest priority level 
+                MUST be treated as the sole heading of the `hgroup`. All other heading elements MUST instead be exposed as if they
+                were `p` elements.
```
Comment (scottaohara) [inline index.html]:
```suggestion
                were <a href=#el-p">`p`</a> elements. See <a class="core-mapping" href="#role-map-paragraph">`paragraph` role on Core AAM</a>.
```

File: index.html [hunk 2]
```diff
@@ -1197,12 +1197,16 @@ <h3>HTML Element Role Mappings</h3>
...
+              <td class="aria"><a class="core-mapping" href="#role-map-group">`group`</a> role </td>
               <td class="ia2"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="uia"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="atk"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="ax"><div class="general">Use WAI-ARIA mapping</div></td>
-              <td class="comments"></td>
+              <td class="comments">
+                If an `hgroup` contains multiple heading elements, then the heading element with the highest priority level 
+                MUST be treated as the sole heading of the `hgroup`. All other heading elements MUST instead be exposed as if they
+                were <a href=#el-p">`p`</a> elements. See <a class="core-mapping" href="#role-map-paragraph">`paragraph` role on Core AAM</a>.
```
Comment (scottaohara) [inline index.html]:
```suggestion
                were <a href="#el-p">`p`</a> elements. See <a class="core-mapping" href="#role-map-paragraph">`paragraph` role on Core AAM</a>.
```

**Comment by stevefaulkner** [comment]
@scottaohara @jnurthen thinking more about mapping, wondering if it is worthwhile having a specific mapping for `hgroup` such as `headinggroup`. Reasoning for this is that if AT would like to provide info to users about `<p>`'s within `hgroup` representing _content as a subheading, subtitle, alternative title or tagline_ with `role=group` this will be ambiguous. If it is specified as a specific type of group it will be clear. I am not thinking that the semantics of the `p` elements or the `<hgroup>` will generally be conveyed to users, but would be good to have a clear mechanism to do so for AT. Thoughts??

File: index.html [hunk 3]
```diff
@@ -1255,13 +1255,17 @@ <h3>HTML Element Role Mappings</h3>
...
+              <td class="aria"><a class="core-mapping" href="#role-map-group">`group`</a> role </td>
               <td class="ia2"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="uia"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="atk"><div class="general">Use WAI-ARIA mapping</div></td>
               <td class="ax"><div class="general">Use WAI-ARIA mapping</div></td>
               <!-- <td class="naming"></td> -->
-              <td class="comments"></td>
+              <td class="comments">
+                If an `hgroup` contains multiple heading elements, then the heading element with the highest priority level 
+                MUST be treated as the sole heading of the `hgroup`. All other heading elements MUST instead be exposed as if they
```
Comment (scottaohara) [inline index.html]:
```suggestion
                MAY be treated as the sole heading of the `hgroup`. All other heading elements MAY instead be exposed as if they
```

File: index.html [hunk 4]
```diff
@@ -1255,13 +1255,17 @@ <h3>HTML Element Role Mappings</h3>
...
+                were <a href="#el-p">`p`</a> elements. See <a class="core-mapping" href="#role-map-paragraph">`paragraph` role on Core AAM</a>.
```
Comment (scottaohara) [inline index.html:1244]:
```suggestion
                were <a href="#el-p">`p`</a> elements. See <a class="core-mapping" href="#role-map-paragraph">`paragraph` role on Core AAM</a>.
                Conformance checkers MUST flag authors for incorrect nesting of multiple heading elements 
                within a single `hgroup`.
```

**Review by scottaohara — COMMENTED** [review]
updates coming out of WG discussion from today

**Comment by scottaohara** [comment]
merging per bugs for this change being filed and no objections to this change from the wg.

**Pull request merged**